packages/io-auth-n-identity-commons/src/types/rejected-login-event.ts

```diff
@@ -0,0 +1,73 @@
+import { IPString, PatternString } from "@pagopa/ts-commons/lib/strings";
+import { enumType } from "@pagopa/ts-commons/lib/types";
+import * as t from "io-ts";
+
+export enum RejectedLoginCauseEnum {
+  AGE_BLOCK = "age_block",
+  AUTH_LOCK = "auth_lock",
+  CF_MISMATCH = "cf_mismatch",
+}
+
+export type RejectedLoginCause = t.TypeOf<typeof RejectedLoginCause>;
+export const RejectedLoginCause = enumType<RejectedLoginCauseEnum>(
+  RejectedLoginCauseEnum,
+  "RejectedLoginCause",
+);
+
+// TODO: Provisional fields (fiscalCode and ts already in the base event)
+export const BaseRejectedLoginEventContent = t.type({
+  // Date of the SPID request / response in YYYY-MM-DD format
+  createdAtDay: PatternString("^[0-9]{4}-[0-9]{2}-[0-9]{2}$"),
+
+  // IP of the client that made a SPID login action
+  ip: t.string.pipe(IPString),
+
+  // XML payload of the SPID Request
+  requestPayload: t.string,
+
+  // XML payload of the SPID Response
+  responsePayload: t.string,
+
+  // SPID request id
+  spidRequestId: t.union([t.undefined, t.string]),
+});
+export type BaseRejectedLoginEventContent = t.TypeOf<
+  typeof BaseRejectedLoginEventContent
+>;
+
+export const AgeBlockRejectedLogin = t.intersection([
+  t.type({
+    rejectionCause: t.literal(RejectedLoginCauseEnum.AGE_BLOCK),
+  }),
+  BaseRejectedLoginEventContent,
+]);
```

Comment:
It would be helpful to include the age limit in effect when the age_block event is triggered, as this value could potentially change in the future.